aiida/cmdline/commands/cmd_export.py

```diff
@@ -141,31 +143,45 @@ def create(
 
 @verdi_export.command('migrate')
 @arguments.INPUT_FILE()
-@arguments.OUTPUT_FILE()
+@arguments.OUTPUT_FILE(required=False)
 @options.ARCHIVE_FORMAT()
 @options.FORCE(help='overwrite output file if it already exists')
+@click.option('-i', '--in-place', is_flag=True, help='Migrate the archive in place, overwriting the original file.')
 @options.SILENT()
 @click.option(
     '-v',
     '--version',
     type=click.STRING,
     required=False,
     metavar='VERSION',
-    help='Specify an exact archive version to migrate to. By default the most recent version is taken.'
+    default=EXPORT_VERSION,
+    help='Archive format version to migrate to.',
+    show_default=True,
 )
-def migrate(input_file, output_file, force, silent, archive_format, version):
+def migrate(input_file, output_file, force, silent, in_place, archive_format, version):
     # pylint: disable=too-many-locals,too-many-statements,too-many-branches
     """Migrate an export archive to a more recent format version."""
     import tarfile
     import zipfile
+    import shutil
 
     from aiida.common import json
     from aiida.common.folders import SandboxFolder
-    from aiida.tools.importexport import EXPORT_VERSION, migration, extract_zip, extract_tar, ArchiveMigrationError
+    from aiida.tools.importexport import migration, extract_zip, extract_tar, ArchiveMigrationError
 
     if version is None:
         version = EXPORT_VERSION
 
+    if in_place:
+        if output_file:
+            echo.echo_critical('output file specified together with --in-place flag')
+        tempdir = tempfile.mkdtemp()
```

Comment:
Minor nit, but I think `tempfile.TemporaryDirectory()` would be better here, even if used without the context manager.

The functional difference is that `tempfile.TemporaryDirectory()` gets cleaned when collected by the garbage collector, for example, if there is an error somewhere in the code between here and the `shutil.rmtree`.

Reply:
Oh, nice, I wasn't aware of this feature. Thanks, will fix!

Reply:
There might be a couple of other places in the AiiDA codebase where we may want to switch from `mkdtemp` to `TemporaryDirectory`. Feel free to open an issue about this.

Reply:
Done, see https://github.com/aiidateam/aiida-core/issues/4336. 

Note that you'll have to remove it using `.cleanup()` instead of `shutil.rmtree` - otherwise it'll try to delete it twice (and fail).